fish_right_prompt.fish

```diff
@@ -24,6 +23,15 @@ function fish_right_prompt
       end
     end
 
-    echo -n -s $ruby_info $normal
+    if [ "$theme_display_rbenv_with_gemfile_only" = 'yes' ]
+      if test -f Gemfile
+        echo -n -s $ruby_info $normal
+      end
+      return
```

Comment:
From what I can see, the last line you call changes the color to red: `set ruby_info $red(_ruby_version)` (either on line 17 or 21). Which means the text typed in the terminal will have this color instead of `normal`.

Before this change, `echo -n -s $ruby_info $normal` was always being executed, so the color would be `normal` instead of `red`.

Following the possible flows on your `if/else`. You are missing a else on `if test -f Gemfile`.